hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java

```diff
@@ -528,6 +529,7 @@ private static QueryDataType resolveType(String columnTypeName) {
             case "INT":
             case "INT4":
             case "INTEGER":
+            case "NUMBER":
                 return QueryDataType.INT;
```

Comment:
Number should convert to particular types based on the precision and scale parameters.